zone/lua_bot.cpp

```diff
@@ -1,107 +1,36 @@
-	Lua_Safe_Call_Class(Lua_Mob);
-	return Lua_Mob(self->GetOwner());
-}
-
-bool Lua_Bot::HasBotItem(uint32 item_id) {
-	Lua_Safe_Call_Bool();
-	return self->HasBotItem(item_id);
-}
-
-void Lua_Bot::RemoveBotItem(uint32 item_id) {
-	Lua_Safe_Call_Void();
-	self->RemoveBotItem(item_id);
-}
-
-Lua_ItemInst Lua_Bot::GetBotItem(uint16 slot_id) {
-	Lua_Safe_Call_Class(Lua_ItemInst);
-	return self->GetBotItem(slot_id);
-}
-
-uint32 Lua_Bot::GetBotItemIDBySlot(uint16 slot_id) {
-	Lua_Safe_Call_Int();
-	return self->GetBotItemBySlot(slot_id);
-}
-
-luabind::scope lua_register_bot() {
-	return luabind::class_<Lua_Bot, Lua_Mob>("Bot")
-	.def(luabind::constructor<>())
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32,uint32,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32,uint32,uint32,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32,uint32,uint32,uint32,uint32))&Lua_Bot::AddBotItem)
-	.def("AddBotItem", (void(Lua_Bot::*)(uint16,uint32,int16,bool,uint32,uint32,uint32,uint32,uint32,uint32))&Lua_Bot::AddBotItem)
-	.def("CountBotItem", (uint32(Lua_Bot::*)(uint32))&Lua_Bot::CountBotItem)
-	.def("GetBotItem", (Lua_ItemInst(Lua_Bot::*)(uint16))&Lua_Bot::GetBotItem)
-	.def("GetBotItemIDBySlot", (uint32(Lua_Bot::*)(uint16))&Lua_Bot::GetBotItemIDBySlot)
-	.def("GetOwner", (Lua_Mob(Lua_Bot::*)(void))&Lua_Bot::GetOwner)
-	.def("HasBotItem", (bool(Lua_Bot::*)(uint32))&Lua_Bot::HasBotItem)
-	.def("RemoveBotItem", (void(Lua_Bot::*)(uint32))&Lua_Bot::RemoveBotItem);
+#include "lua_entity.h"
+
+void lua_register_bot(sol::state_view &sv)
+{
+	auto bot = sv.new_usertype<Lua_Bot>("Bot", sol::constructors<Lua_Bot()>(), sol::base_classes,
+					    sol::bases<Lua_Mob, Lua_Entity>());
+	bot["AddBotItem"] = sol::overload(
+	    (void(Lua_Bot::*)(uint16, uint32)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool, uint32)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool, uint32, uint32)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool, uint32, uint32, uint32)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool, uint32, uint32, uint32, uint32)) & Lua_Bot::AddBotItem,
+	    (void(Lua_Bot::*)(uint16, uint32, int16, bool, uint32, uint32, uint32, uint32, uint32)) &
+		Lua_Bot::AddBotItem,
```

Comment:
I think we should compact exports onto single lines for readability even if they're long. Also there's spaces around `&` on these

Reply:
I just used clang-format to format these

Reply:
Definitely agree with the one line